scipy/optimize/tests/test_bracket.py

```diff
@@ -286,15 +286,387 @@ def f(x):
+        assert result.success
+        assert result.nfev == f.count
+
+    @pytest.mark.parametrize(
+        "xl0,xm0,xr0,xmax",
+        (
+            (0.2, 0.3, 0.4, 1.0),
+            (0.05, 0.075, 0.1, 1.0),
+            (-0.2, -0.1, 0.0, 1.0),
+            (-21.2, -17.7, -14.2, 1.0),
+            (0.2, 0.3, None, 1.0),
+            (0.05, 0.075, None, 1.0),
+            (-0.2, -0.1, None, 1.0),
+            (-21.2, -17.7, None, 1.0),
+        )
+    )
+    @pytest.mark.parametrize(
+        "args", (
+            (0.9999999999999999, 0.0), (0.9, 0.0), (0.7, 0.0), (0.5, 0.0)
+        )
+    )
+    def test_scalar_with_limit_right(self, xl0, xm0, xr0, xmax, args):
+        f = self.init_f()
+        kwargs = self.get_kwargs(xl0=xl0, xr0=xr0, xmax=xmax, args=args)
+        result = _bracket_minimum(f, xm0, **kwargs)
+        self.assert_valid_bracket(result)
+        assert result.status == 0
+        assert result.success
+        assert result.nfev == f.count
+
+    @pytest.mark.parametrize(
+        "xl0,xm0,xr0,xmin,xmax,args",
+        (
+            (0.2, 0.3, 0.4, None, 1.0, (1.0, 0.0)),
+            (1.4, 1.95, 2.5, 0.3, None, (0.3, 0.0)),
+            (2.6, 3.25, 3.9, None, 99.4, (99.4, 0)),
+            (4, 4.5, 5, -26.3, None, (-26.3, 0)),
+            (None, 0.3, None, None, 1.0, (1.0, 0.0)),
+            (None, 1.95, None, 0.3, None, (0.3, 0.0)),
+            (None, 3.25, None, None, 99.4, (99.4, 0)),
+            (None, 4.5, None, -26.3, None, (-26.3, 0)),
+        )
+    )
+    def test_minima_at_boundary_point(self, xl0, xm0, xr0, xmin, xmax, args):
+        f = self.init_f()
+        kwargs = self.get_kwargs(xr0=xr0, xmin=xmin, xmax=xmax, args=args)
+        result = _bracket_minimum(f, xm0, **kwargs)
+        assert result.status == -1
+        assert args[0] in (result.xl, result.xr)
+        assert result.nfev == f.count
+
+    @pytest.mark.parametrize(
+        "xm0",
+        (
+            np.array([[0.55], [0.58]]),
+            np.array([[0.55, 0.56], [0.57, 0.58]]),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "xl0",
+        (
+            0.2,
+            np.array([[0.2], [0.3]]),
+            np.array([[0.2, 0.3],
+                      [0.3, 0.4]])
+        ),
+    )
+    @pytest.mark.parametrize(
+        "xr0",
+        (
+            0.6,
+            np.array([[0.6], [0.8]]),
+        ),
+    )
+    @pytest.mark.parametrize(
+        "xmin",
+        (
+            np.array([[-np.inf], [-1.0]]),
+            np.array([[-np.inf, -1.0], [-2.0, -3.0]]),
+        )
+    )
+    @pytest.mark.parametrize(
+        "xmax",
+        (
+            np.array([[np.inf], [2.0]]),
+            np.array([[np.inf, 2.0], [3.5, 4.5]]),
+        )
+    )
+    @pytest.mark.parametrize(
+        "args",
+        (
+            (0.0, 0.0),
+            (np.array([[0.0], [-0.1]]), np.array([[0.0], [2.5]])),
+        )
+    )
+    def test_vectorized(self, xl0, xm0, xr0, xmin, xmax, args):
```

Comment:
Most of the other tests followed the pattern of `TestBracketRoot` pretty carefully, and we already decided that those were acceptable, so it's easy to judget that these are acceptable by analogy. This test is quite different - why did you decide to change this one? Is it as strong or stronger in your opinion?

Reply:
This is one of the first tests I wrote before looking at the tests for `TestBracketRoot`, and I just neglected to change it after your review feedback. I don't think it's stronger, but it was useful while I was implementing things. I'll change it to match the convention used in `TestBracketRoot`.